src/ros_filter.cpp

```diff
@@ -2243,6 +2257,148 @@ bool RosFilter<T>::setPoseSrvCallback(
   return true;
 }
 
+template<typename T>
+void RosFilter<T>::setStateCallback(const robot_localization::msg::State::SharedPtr msg)
```

Comment:
I feel like this duplicates a lot of code. Why can't we just call [reset()](https://github.com/cra-ros-pkg/robot_localization/blob/69643e19cbfcc2a1e01836b78f416753fab1a627/src/ros_filter.cpp#L125-L153) and then [set the state](https://github.com/cra-ros-pkg/robot_localization/blob/69643e19cbfcc2a1e01836b78f416753fab1a627/src/ros_filter.cpp#L987-L999)? As much as I appreciate the input sanitising, I think the user needs to be ensuring that they are sending valid states to the filter. This class is already too bloated, and this method adds a lot of new stuff.